File: vaultx-frontend/src/services/bundleService.js
import api from './api';

const bundleService = {
  // Create a new bundle
  createBundle: (data) => {
    return api.post('/bundles', data);
  },

  // Get user bundles (dashboard)
  getUserBundles: (page = 0, size = 10, sort = 'updatedAt', dir = 'desc') => {
    return api.get('/bundles', { params: { page, size, sort, dir } });
  },

  // Get a specific bundle
  getBundle: (id) => {
    return api.get(`/bundles/${id}`);
  },

  // Update bundle details
  updateBundle: (id, data) => {
    return api.put(`/bundles/${id}`, data);
  },

  // Duplicate a bundle
  duplicateBundle: (id) => {
    return api.post(`/bundles/${id}/duplicate`);
  },

  // Toggle favourite status
  toggleFavourite: (id) => {
    return api.post(`/bundles/${id}/favorite`);
  },

  // Archive bundle
  archiveBundle: (id) => {
    return api.post(`/bundles/${id}/archive`);
  },

  // Restore bundle
  restoreBundle: (id) => {
    return api.post(`/bundles/${id}/restore`);
  },

  // Delete bundle
  deleteBundle: (id) => {
    return api.delete(`/bundles/${id}`);
  },

  // Add documents to bundle
  addDocuments: (id, documentIds) => {
    return api.post(`/bundles/${id}/documents`, { documentIds });
  },

  // Remove document from bundle
  removeDocument: (id, documentId) => {
    return api.delete(`/bundles/${id}/documents/${documentId}`);
  },

  // Reorder documents
  reorderDocuments: (id, documentIds) => {
    return api.put(`/bundles/${id}/reorder`, { documentIds });
  },

  // Download bundle as ZIP
  downloadBundleAsZip: (id) => {
    return api.get(`/bundles/${id}/download`, { responseType: 'blob' });
  }
};

export default bundleService;

File: vaultx-backend/src/main/java/com/vaultx/service/BundleService.java
package com.vaultx.service;

import com.vaultx.common.PagedResponse;
import com.vaultx.dto.bundle.BundleCreateRequest;
import com.vaultx.dto.bundle.BundleUpdateRequest;
import com.vaultx.dto.bundle.BundleDto;
import com.vaultx.dto.bundle.DocumentReorderRequest;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface BundleService {
    BundleDto createBundle(BundleCreateRequest request);
    BundleDto updateBundle(UUID id, BundleUpdateRequest request);
    PagedResponse<BundleDto> getUserBundles(Pageable pageable);
    BundleDto getBundleById(UUID id);
    BundleDto duplicateBundle(UUID id);
    BundleDto toggleFavourite(UUID id);
    BundleDto archiveBundle(UUID id);
    BundleDto restoreBundle(UUID id);
    void deleteBundle(UUID id);
    byte[] downloadBundle(UUID id);
    
    BundleDto addDocuments(UUID bundleId, List<UUID> documentIds);
    BundleDto removeDocument(UUID bundleId, UUID documentId);
    BundleDto reorderDocuments(UUID bundleId, DocumentReorderRequest request);
}
